### DISCUSSION TITLE
Problem with color display after usage of unix "screen" command

### DISCUSSION THREAD

**Discussion by Kauzge** [JanDeDobbeleer/oh-my-posh#556] [Troubleshoot]
I followed the installation instructions on the documentation website both for my Windows machine (Windows Terminal) and Ubuntu server (version 20.04.2 LTS).
Yesterday everything worked fine. But today after I logged into my server and used the "screen" command the colors didn't get displayed correctly (see image below).
I'm using the Caskaydia Nerd Font.

Did I make a mistake or is this a bug?

![Oh-my-posh](https://user-images.githubusercontent.com/61211374/111909091-2bb77600-8a5c-11eb-92c0-47c9e4761744.png)
(Splitted panes --> Left side: Ubuntu after login | Right side: Ubuntu after "screen" command)

**Comment by JanDeDobbeleer** [comment]
Definitely not an issue on your end, and also not on ours. I'll check the Windows Terminal issue list 😅

**Reply by Kauzge** [reply]
I tried to multiplex my terminal with the "tmux" command instead of "screen". With tmux everything gets displayed properly. Maybe there is a problem within the screen environment?

**Comment by andrzejnovak** [comment]
I've ran into the same problem (I am also using windows terminal). 

![image](https://user-images.githubusercontent.com/13226500/158537217-70d9ac5d-4727-4110-95f4-b6c39296fb04.png)
![image](https://user-images.githubusercontent.com/13226500/158537319-b0cb0ec9-7b0f-4c23-83b3-4ee7193f3616.png)

Funnily enough, I've tried switching to oh-my-posh after I couldn't fix issues with https://github.com/justjanne/powerline-go that has text wrapping problems and duplicates prompt at times when the commands/entries in history got too long. So far I haven't noticed that in oh-my-posh (yay 🎉 ) but it didn't suffer from the screen problem. Not sure how similar the rendering engine is between the two.

**Comment by ghost** [comment]
Is there any fix for this?
I am seeing this issue as well.

I am on WSL2 and trying to use OMP through "screen" from Windows Terminal

**Comment by ZhiZe-ZG** [comment]
I have the same problem when using alacritty as terminal.

The color of oh-my-posh have problem but the neofetch works well.

I've tried the ANSI escape sequences. Basic colors works well but the true color mode can not show properly. It seems like the GNU Screen do not support the true color mode.

![Screenshot 2024-08-24 070548](https://github.com/user-attachments/assets/96cc8493-8129-4412-8988-c121ab268b11)
![Screenshot 2024-08-24 071048](https://github.com/user-attachments/assets/052540cd-462f-41b8-a0c3-452a9a3be1cb)

**Comment by MalumaDev** [comment]
Any fix?

**Reply by JanDeDobbeleer** [reply]
@MalumaDev not on our end. This is something screen/tmux should look at. We output the same.

**Reply by ZhiZe-ZG** [reply]
Try tmux.

**Comment by WizardCM** [comment]
I realise this is a bit of a necro, but as this is the first Google result for this query, I thought I'd post the answer:

Add the following to your `.screenrc`, at least in screen v5 (released Aug 2024):

```
truecolor on
```

And voila!

If you don't have v5 on your system, install it via `linuxbrew`.

Basically, after a ton of troubleshooting, I determined oh-my-posh uses RGB, while `screen` was only rendering up to 256 colours. All environment variables looked correct, so I began digging and ended up [here](https://github.com/derailed/k9s/issues/2810#issuecomment-2251739737).

**Reply by JanDeDobbeleer** [reply]
Nothing like a bit of necromancy. Thanks for the answer!

**Reply by ZhiZe-ZG** [reply]
I recently asked GPT this problem, it also attributed this to the color mode of screen/tmux. Thanks for your answer.
